generators/server/templates/dotnetcore/src/Project/Models/User.cs.ejs

```diff
@@ -96,5 +97,103 @@ namespace <%= namespace %>.Models {
                    $", ActivationKey='{ActivationKey}'" +
                    "}";
         }
+        
+        public static SortExpressions<User> GetSortExpressions(Sort sort)
+        {
+            var sortExpressions = new SortExpressions<User>();
+            if (sort != null && sort.IsSorted())
+            {
+                IList<Order> orders = sort.Orders;
+                foreach (var order in orders)
+                {
+                    AddSortExpression(sortExpressions, order);                    
+                }
+
+            }
+            return sortExpressions;
+        }
+
+        private static void AddSortExpression(SortExpressions<User> sortExpressions, Order order)
```

Comment:
Have you test reflection performance here ? Something like : `PropertyInfo[] properties = typeof(User).GetProperties();` to avoid writing too much code. Maybe even write a completely generic code for other class

Reply:
No, I didn't try to test. Do you have a proposal ?

Reply:
something like this 
```public static IQueryable<TEntity> OrderBy<TEntity>(this IQueryable<TEntity> source, string orderByValues) where TEntity : class
{
  IQueryable<TEntity> returnValue = null;

  string orderPair = orderByValues.Trim().Split(',')[0];
  string command = orderPair.ToUpper().Contains("DESC") ? "OrderByDescending" : "OrderBy";

  var type = typeof(TEntity);
  var parameter = Expression.Parameter(type, "p");

  string propertyName = (orderPair.Split(' ')[0]).Trim();

  System.Reflection.PropertyInfo property;
  MemberExpression propertyAccess;

  if (propertyName.Contains('.'))
  {
    // support to be sorted on child fields. 
    String[] childProperties = propertyName.Split('.');
    property = typeof(TEntity).GetProperty(childProperties[0]);
    propertyAccess = Expression.MakeMemberAccess(parameter, property);

    for (int i = 1; i < childProperties.Length; i++)
    {
      Type t = property.PropertyType;
      if (!t.IsGenericType)
      {
        property = t.GetProperty(childProperties[i]);
      }
      else
      {
        property = t.GetGenericArguments().First().GetProperty(childProperties[i]);
      }

      propertyAccess = Expression.MakeMemberAccess(propertyAccess, property);
    }
  }
  else
  {
    property = type.GetProperty(propertyName);
    propertyAccess = Expression.MakeMemberAccess(parameter, property);
  }

  var orderByExpression = Expression.Lambda(propertyAccess, parameter);

  var resultExpression = Expression.Call(typeof(Queryable), command, new Type[] { type, property.PropertyType },

  source.Expression, Expression.Quote(orderByExpression));

  returnValue = source.Provider.CreateQuery<TEntity>(resultExpression);

  if (orderByValues.Trim().Split(',').Count() > 1)
  {
    // remove first item
    string newSearchForWords = orderByValues.ToString().Remove(0, orderByValues.ToString().IndexOf(',') + 1);
    return source.OrderBy(newSearchForWords);
  }

  return returnValue;
}```


source : https://stackoverflow.com/questions/307512/how-do-i-apply-orderby-on-an-iqueryable-using-a-string-column-name-within-a-gene

Reply:
other link : https://code-maze.com/sorting-aspnet-core-webapi/

Reply:
with that we will have even more code. By cons, I don't understand, you propose that for what part of the implementation that I proposed!

For me, it is too complicated to propose a 100% generic solution for sorting because it really depends on the properties of each entity to generate with JDL. This is why I added this method for each entity in Entity.cs.ejs in addition to the User.cs class:

The generic part that I proposed is in the class SortExpressions.cs.ejs and QueryableExtensions.cs.ejs

```
public static SortExpressions<<%= entityClassName %>> GetSortExpressions(Sort sort)
        {
            var sortExpressions = new SortExpressions<<%= entityClassName %>>();
            if (sort != null && sort.IsSorted())
            {
                IList<Order> orders = sort.Orders;
                foreach (var order in orders)
                {
                    AddSortExpression(sortExpressions, order);                    
                }

            }
            return sortExpressions;
        }

        private static void AddSortExpression(SortExpressions<<%= entityClassName %>> sortExpressions, Order order)
        {
            if (sortExpressions == null || order == null || order.Property == null) return;
            bool isDescending = order.Direction.IsDescending();
            switch (order.Property.ToLower())
            {
                case "id":
                    {
                        sortExpressions.Add(x => x.Id, isDescending);
                        break;
                    }

                <%_ for (idx in fields) {
                    const fieldNamePascalized = fields[idx].fieldNamePascalized; _%>
                case "<%= fieldNamePascalized.toLowerCase() %>":
                    {
                        sortExpressions.Add(x => x.<%= fieldNamePascalized %>, isDescending);
                        break;
                    }
                <%_ } _%>

                default:
                    break;
            }
        }
```

Reply:
done with reflection performance:

```
private static IQueryable<TEntity> ApplySort<TEntity>(this IQueryable<TEntity> query, Sort sort)
        {
            if (!query.Any() || sort == null || sort.IsUnsorted()) return query;

            var sortExpressions = new SortExpressions<TEntity>();
            var propertyInfos = typeof(TEntity).GetProperties(BindingFlags.Public | BindingFlags.Instance);

            IList<Order> orders = sort.Orders;
            foreach (var order in orders)
            {
                if (order == null || order.Property == null) continue;
                
                bool isDescending = order.Direction.IsDescending();
                var objectProperty = propertyInfos.FirstOrDefault(pi => pi.Name.Equals(order.Property, StringComparison.InvariantCultureIgnoreCase));
                if (objectProperty == null) continue;

                var type = typeof(TEntity);
                var parameter = Expression.Parameter(type, "p");
                var property = type.GetProperty(objectProperty.Name);
                var propertyAccess = Expression.MakeMemberAccess(parameter, property);
                var expressionFunc = Expression.Lambda<Func<TEntity, object>>(propertyAccess, parameter);

                sortExpressions.Add(expressionFunc, isDescending);
            }
            return SortExpressions<TEntity>.ApplySorts(query, sortExpressions);
        }
```